### PR TITLE
[Merged by Bors] - Add info log to v2alpha1 SubmitTransaction method

### PR DESCRIPTION
**Pull request by kacpersaw**
## Description

Add a info log with the transaction ID after submitting the transaction.

### CODE REVIEW

**Comment by pigmej** [comment]
We need different way to debug that. We cannot enable globally `debug` as that will enable all debug logs, isn't it?

**Comment by jellonek** [comment]
We can configure specific logger for grpc service as stated in https://github.com/spacemeshos/go-spacemesh/blob/develop/config/logging.go#L48 but I'm not sure what will be the impact.

File: api/grpcserver/v2alpha1/transaction.go [hunk 1]
```diff
@@ -207,6 +209,9 @@ func (s *TransactionService) SubmitTransaction(
 	}
 
 	raw := types.NewRawTx(request.Transaction)
+	ctxzap.Info(ctx, "successfully submitted transaction",
+		zap.String("tx_id", raw.ID.String()),
```
Comment (poszu) [inline api/grpcserver/v2alpha1/transaction.go]:
```suggestion
		zap.Stringer("tx_id", raw.ID),
```

**Review by poszu — CHANGES_REQUESTED** [review]

**Review by poszu — APPROVED** [review]

**Comment by poszu** [comment]
> We can configure specific logger for grpc service as stated in [`develop`/config/logging.go#L48](https://github.com/spacemeshos/go-spacemesh/blob/develop/config/logging.go?rgh-link-date=2024-11-12T15%3A48%3A33Z#L48) but I'm not sure what will be the impact.

It is already using a logger configured with `GrpcLoggerLevel`. It is configured via middleware here: https://github.com/spacemeshos/go-spacemesh/blob/f711c614a47c7cabcfa886cea263df01dfb98110/api/grpcserver/grpc.go#L138

**Comment by kacpersaw** [comment]
bors merge

**Pull request closed without merging**